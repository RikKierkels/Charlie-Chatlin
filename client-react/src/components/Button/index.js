import React from 'react';
import styled from 'styled-components';

const Button = ({ type = 'submit', children, ...props }) => {
  return (
    <StyledButton type={type} {...props}>
      {children}
    </StyledButton>
  );
};

export default Button;

const StyledButton = styled.button`
  width: 3.5rem;
  height: 3.5rem;
  border: 0;
  border-radius: 50%;
  background-color: inherit;
  color: inherit;
  cursor: pointer;
`;
