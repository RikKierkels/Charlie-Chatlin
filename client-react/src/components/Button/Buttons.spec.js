import React from 'react';
import { render } from '@testing-library/react';
import Button from './index';

test("renders it's children", () => {
  const { container } = render(
    <Button>
      <span>Hello World!</span>
    </Button>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
