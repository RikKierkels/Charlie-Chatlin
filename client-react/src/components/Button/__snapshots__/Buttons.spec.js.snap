// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders it's children 1`] = `
<button
  class="sc-AxjAm hWhgq"
  type="submit"
>
  <span>
    Hello World!
  </span>
</button>
`;
